msticpy/nbtools/timeseries.py

```diff
@@ -0,0 +1,260 @@
+# pylint: disable=too-many-branches, too-many-function-args, too-many-arguments
+@export  # noqa: C901, MC0001
+def display_timeseries_anomolies(
+    data: pd.DataFrame,
+    y: str = "Total",
+    time_column: str = "TimeGenerated",
+    anomalies_column: str = "anomalies",
+    source_columns: list = None,
+    period: int = 30,
+    **kwargs
+) -> figure:
+    """
+    Display time series anomalies visualization.
+
+    Parameters
+    ----------
+    data : pd.DataFrame
+        DataFrame as a time series data set retreived from KQL time series functions
+        dataframe will have columns as TimeGenerated, y, baseline, score, anomalies
+    y : str, optional
+        Name of column holding numeric values to plot against time series to determine anomolies
+        (the default is 'Total')
+    time_column : str, optional
+        Name of the timestamp column
+        (the default is 'TimeGenerated')
+    anomalies_column : str, optional
+        Name of the column holding binary status(1/0) for anomaly/benign
+        (the default is 'anomolies')
+    source_columns : list, optional
+        List of default source columns to use in tooltips
+        (the default is None)
+    period : int, optional
+        Period of the dataset for hourly-no of days, for daily-no of weeks.
+        This is used to correctly calculate the plot height.
+        (the default is 30)
+
+    Other Parameters
+    ----------------
+    ref_time : datetime, optional
+        Input reference line to display (the default is None)
+    title : str, optional
+        Title to display (the default is None)
+    legend: str, optional
+        Where to position the legend
+        None, left, right or inline (default is None)
+    yaxis : bool, optional
+        Whether to show the yaxis and labels
+    range_tool : bool, optional
+        Show the the range slider tool (default is True)
+    height : int, optional
+        The height of the plot figure
+        (the default is auto-calculated height)
+    width : int, optional
+        The width of the plot figure (the default is 900)
+
+    Returns
+    -------
+    figure
+        The bokeh plot figure.
+
+    """
+    check_kwargs(kwargs, _DEFAULT_KWARGS + _TL_VALUE_KWARGS)
+
+    reset_output()
+    output_notebook()
+    height: int = kwargs.pop("height", None)
+    width: int = kwargs.pop("width", 1200)
+    title: str = kwargs.pop("title", None)
+    time_column = kwargs.get("x", time_column)
+    show_range: bool = kwargs.pop("range_tool", True)
+    color: list = kwargs.get("color", ["navy", "green", "firebrick"])
+    legend_pos: str = kwargs.pop("legend", "top_left")
+    xgrid: bool = kwargs.pop("xgrid", False)
+    ygrid: bool = kwargs.pop("ygrid", False)
+    kind: str = kwargs.pop("kind", "circle_x")
+
+    ref_time, ref_label = _get_ref_event_time(**kwargs)
+
+    # Manually append or set default color scheme if user provides one or two colors
+    if len(color) == 1:
```

Comment:
Doesn't this mean if I specify 3 colors, they will be ignored?

Hmm - had a thought and did an experiment:
from itertools import zip_longest
color = [col1 or col2 for col1, col2 in zip_longest(color[:3], default_colors)]

Assuming default_colors is your default list and color is the user-specified list. This should iterate through them (max 3 items from color and all of default) and select col1 if it is not None and col2 otherwise.
:-) No obligation to take the code - I was just intrigued if there was another way to do it.